front-end/src/app/layout/layout.component.ts

```diff
@@ -1,23 +1,73 @@
-import { Component, OnInit } from '@angular/core';
-import { Observable } from 'rxjs';
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { Observable, Subject, takeUntil } from 'rxjs';
 import { Store } from '@ngrx/store';
 import { selectCommitteeAccount } from '../store/committee-account.selectors';
 import { selectSpinnerStatus } from '../store/spinner.selectors';
 import { CommitteeAccount } from 'app/shared/models/committee-account.model';
+import { NavigationEnd, Router, Event } from '@angular/router';
 
 @Component({
   selector: 'app-layout',
   templateUrl: './layout.component.html',
-  styleUrls: ['./layout.component.scss']
+  styleUrls: ['./layout.component.scss'],
 })
-export class LayoutComponent implements OnInit {
+export class LayoutComponent implements OnInit, OnDestroy {
   committeeAccount$: Observable<CommitteeAccount> | undefined;
   progressBarVisible$: Observable<{ spinnerOn: boolean }> | undefined;
+  sidebarVisible = false;
+  private destroy$ = new Subject<boolean>();
 
-  constructor(private store: Store) {}
```

Comment:
For this component and for the menu component, it would be cool if we assign `sidebarState` with nested `sidebarVisible` and `expandedSection` in with a resolver.  We could apply the resolver in *-routing.module.ts like so:
```

  {
    path: '',
    title: 'Manage Reports',
    component: ReportListComponent,
    pathMatch: 'full',
  },
  ...
  {
    path: 'f3x/summary/:reportId',
    title: 'View summary page',
    component: ReportSummaryComponent,
    resolve: { report: ReportResolver, sidebarState: SidebarResolver(MenuSections.REVIEW) },
  },
```

That way, there'd be no regexes and the configurations would be applied once per route, at the route config, instead again every time and everywhere we want to check.  Less router subscriptions and event handlers too

Not a blocker, just a thought seeing as we had to duplicate the url regexes

Reply:
@Elaine-Krauss-TCG @mjtravers Thoughts?

Reply:
I really like Todd's idea of moving the logic into a resolver that then is attached (or not) to the routing. One thing to take into account is that there are 2 things to hide/show here: (1) is the left had blue area which is basically the FECFile Online header and blue area (aka the sidebar component) and (2) the Report Progress menu (aka menu-report component). It could be in the future that the sidebar is visible but a menu component other than the menu-report component will be visible in the sidebar. The resolver solution should be flexible enough to account for different components being inserted into the sidebar component (we do something similar in the transaction-container component)